app/Repository/Group/Group.php

```diff
@@ -0,0 +1,185 @@
+     * @return Member[]
+     */
+    public function getAllMembers() {
+        //ToDo
+        return [];
+    }
+
+    /**
+     * Calculates the root path
+     * @param $groupRepository GroupRepository
+     * @return int[]
+     * @throws WeblingAPIException
+     * @throws \App\Exceptions\GroupNotFoundException
+     * @throws \Webling\API\ClientException
+     */
+    public function calculateRootPath($groupRepository): array
+    {
+        //ToDo
+        if($this->parent === null) {
+            $this->rootPath = [];
+        } else {
+            $parentObject = $groupRepository->get($this->parent);
+
+            $this->rootPath = $parentObject->getRootPath($groupRepository);
+            $this->rootPath[] = $this->parent;
+        }
+
+        return $this->rootPath;
+    }
+
+    /**
+     * @param GroupRepository $groupRepository
+     * @return int[]
+     * @throws WeblingAPIException
+     * @throws \App\Exceptions\GroupNotFoundException
+     * @throws \Webling\API\ClientException
+     */
+    public function getRootPath(GroupRepository $groupRepository): array
+    {
+        if($this->rootPath == null) {
+            $this->calculateRootPath($groupRepository);
+        }
+
+        return $this->rootPath;
+    }
+
+    /**
+     * @param int $id
+     */
+    public function setId(int $id): void
+    {
+        $this->id = $id;
+    }
+
+    /**
+     * @param string $name
+     */
+    public function setName(string $name): void
+    {
+        $this->name = $name;
+    }
+
+    /**
+     * @param int|null $parent
+     */
+    public function setParent(?int $parent): void
+    {
+        $this->parent = $parent;
+    }
+
+    /**
+     * @param Group[] $children
+     */
+    public function setChildren(array $children): void
+    {
+        $this->children = $children;
+    }
+
+    /**
+     * @param Member[] $members
+     */
+    public function setMembers(array $members): void
+    {
+        $this->members = $members;
+    }
+
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @return Group[]
+     */
+    public function getChildren(): array
+    {
+        return $this->children;
+    }
+
+    /**
+     * @return Member[]
+     */
+    public function getMembers(): array
+    {
+        return $this->members;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getParent(): ?int
+    {
+        return $this->parent;
+    }
+
+    /**
+     * Specify data which should be serialized to JSON
+     * @link https://php.net/manual/en/jsonserializable.jsonserialize.php
+     * @return mixed data which can be serialized by <b>json_encode</b>,
+     * which is a value of any type other than a resource.
+     * @since 5.4.0
+     */
+    public function jsonSerialize()
+    {
+        $array =  get_object_vars($this);
+        foreach ($array as $key => $value) {
+            if($value == null) {
```

Comment:
In diesem Fall zwar nicht wichtig, aber ich empfehle grundsätzlich === statt == zu verwenden. === ist schneller und zusätzlich weniger Fehleranfällig, da keine type coersion gemacht wird.

Reply:
Macht Sinn, ja. In meinem Kopf ist == noch default ...